tcplistener.go

```diff
@@ -148,6 +147,18 @@ func handleListenedConn(address string, conn *net.TCPConn, headerByteSize int, m
 	headerBuffer := make([]byte, headerByteSize)
 	dataBuffer := make([]byte, maxMessageSize)
 	for {
+		// Before we try reading a message, check to see if we've been shutdown or not
+		// That way, shutting down will allow any current message to finish, but block
+		// any receives on the connection after it occurs, in addition to blocking new
+		// connections from being accepted, above in the initial connection code.
+		select {
+		case <-sdChan:
+			conn.Close()
+			return
+		default:
+			// Nothing, continue on to accept incoming bytes
+		}
+
```

Comment:
To do that, i'd have to keep a reference to this connection somewhere inside of the listener, and I've explicitly avoided doing that. Once an incoming connection has been received, it's handed to the handler, and only kept around inside of there.

Having the Listener hold state would be a slightly larger change than I feel the scope of this bug fix entails, that would be more of a re-design of the listener - which I'm not opposed to, but my main thrust here is to fix the bug as it exists today, if that makes sense?

I'll read up on the atomic idea, I haven't played around with that package yet.